issue-1 trying to simplify...

ktonegawa opened the pull request:
This is to address issue https://github.com/ktonegawa/vfx_related_scripts/issues/1

Things done:

- Replaced `MArrayDataHandle` with `MDataHandle`
- Tried to set up for envelope attribute
- Removed `MFnMesh` because it was not needed (already have the `MItGeometry` as an input)
- Changing the "print" function from `MGlobal::displayInfo()` to `cerr`

ktonegawa committed:
issue-1 trying to simplify...

dansheerin commented on blendMeshTestDeformer/src/blendMeshDeformer.cpp:
```diff
@@ -1,4 +1,5 @@
+#include <iostream>
 
 MTypeId BlendMesh::id(0x00000233);
 MObject BlendMesh::aNoiseValue;
```
Not needed in simple version

dansheerin commented on blendMeshTestDeformer/src/blendMeshDeformer.cpp:
```diff
@@ -1,4 +1,5 @@
+#include <iostream>
 
 MTypeId BlendMesh::id(0x00000233);
 MObject BlendMesh::aNoiseValue;
```
Not needed in simple version

ktonegawa replied:
Hi Dan, don't I need to include this though to make sure the `cerr` command works?

dansheerin commented on blendMeshTestDeformer/src/blendMeshDeformer.cpp:
```diff
@@ -13,27 +14,20 @@ BlendMesh::~BlendMesh()
-	CHECK_MSTATUS_AND_RETURN_IT(status);
-	MFloatVectorArray normals;
-	fnMesh.getVertexNormals(false, normals);
+	if (MS::kSuccess != status) return status;
```
No need for this line. The Marco above does this already.

dansheerin commented on blendMeshTestDeformer/src/blendMeshDeformer.cpp:
```diff
@@ -47,17 +41,17 @@ MStatus BlendMesh::deform(MDataBlock& data, MItGeometry& itGeo,
 			str = str + data_array[i];
 		}
-		MGlobal::displayInfo("position: " + str);
+		cerr << "position: " << str << endl;
```
I think you can just do:
cerr << "position: " << point << endl;

dansheerin replied:
Remove everything to do with str

dansheerin commented on blendMeshTestDeformer/src/blendMeshDeformer.cpp:
```diff
@@ -47,17 +41,17 @@ MStatus BlendMesh::deform(MDataBlock& data, MItGeometry& itGeo,
 			str = str + ", ";
 			str = str + data_array[i];
```
Not needed

dansheerin commented on blendMeshTestDeformer/src/blendMeshDeformer.cpp:
```diff
@@ -47,17 +41,17 @@ MStatus BlendMesh::deform(MDataBlock& data, MItGeometry& itGeo,
 			str = str + ", ";
 			str = str + data_array[i];
 		}
```
None of L36-L43 are needed as they are just processing to get the point into a str.

ktonegawa committed:
addressing given notes

ktonegawa commented:
- Removed `include <iostream>`
- Removed redundant `if (MS::kSuccess != status) return status;` line
- Removed block that processes `point.get()` into an array

dansheerin reviewed (changes requested):
Looking good. You are still going to need iostream

ktonegawa committed:
adding back iostream and specifying namespace for cerr

ktonegawa commented:
- Added back `<iostream>`
- Changed some variable names to be more obvious
- Specified the namespace which `cerr` should be called from
